docs/rst/tutorials/core/qos/xml_profiles/xml_profiles.rst

```diff
@@ -0,0 +1,400 @@
+  with attribute ``is_default_profile="true"``.
+
+
+Creating clients with different profiles
+........................................
+
+ROS 2 clients contain a publisher to send requests, and a subscription to receive the service's replies.
+*rmw_fastrtps* allows for configuring each of these endpoints separately in the following manner:
+
+* To configure the requests publisher, define a ``<data_writer>`` profile with attribute ``profile_name=topic_name``,
+  where ``topic_name`` is the name of the service after mangling.
+  If such profile is not defined, *rmw_fastrtps* attempts to load a ``<data_writer>`` profile
+  with attribute ``profile_name="client"``.
+  If neither of the previous profiles exist, *rmw_fastrtps* attempts to load the ``<data_writer>`` profile
+  with attribute ``is_default_profile="true"``.
+
+* To configure the reply subscription, define a ``<data_reader>`` profile with ``attribute profile_name=topic_name``,
+  where ``topic_name`` is the name of the service after mangling.
+  If such profile is not defined, *rmw_fastrtps* attempts to load a ``<data_reader>`` profile
+  with attribute ``profile_name="client"``.
+  If neither of the previous profiles exist, *rmw_fastrtps* attempts to load the ``<data_reader>`` profile
+  with attribute ``is_default_profile="true"``.
+
+
+Creating ROS contexts and nodes
+...............................
+
+ROS *context* and *node* entities are mapped to *Fast DDS* Participant entity,
+according to the following table:
+
+
++--------------+--------------------------------------+----------------------------------------+
+| ROS entity   | *Fast DDS* entity since *Foxy*       | *Fast DDS* entity in *Eloquent & below*|
++==============+======================================+========================================+
+| Context      | Participant                          | *Not DDS direct mapping*               |
++--------------+--------------------------------------+----------------------------------------+
+| Node         | *Not DDS direct mapping*             | Participant                            |
++--------------+--------------------------------------+----------------------------------------+
+
+This means that on *Foxy* and later releases, contexts can be configured using a ``<Participant>`` profile
+with attribute ``is_default_profile="true"``.
+The same profile will be used in *Eloquent* and below to configure nodes.
+
+For example, a profile for a ROS 2 context on *Foxy* and later releases would be specified as:
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8" ?>
+    <profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
+        <participant profile_name="participant_profile_ros2" is_default_profile="true">
+            <rtps>
+                <name>profile_for_ros2_context</name>
+            </rtps>
+        </participant>
+    </profiles>
+
+.. _vulcanexus_example:
+
+Example
```

Comment:
This also contains profiles for the services demo. Either we remove them, or we add the example for them as a example of profiles for services (I'd rather see the latter).